base/src/com/google/idea/blaze/base/sync/actions/IncrementalSyncProjectAction.java

```diff
@@ -48,7 +48,7 @@ protected void runSync(Project project, AnActionEvent e) {
   }
 
   public static void doIncrementalSync(Class<?> klass, Project project, @Nullable AnActionEvent e) {
-    if (Blaze.getProjectType(project) == ProjectType.QUERY_SYNC) {
+    if (Blaze.getProjectTypeBeforeSync(project) == ProjectType.QUERY_SYNC) {
```

Comment:
So when I switch from legacy sync to aspect sync, QS is able to detect it's the first sync and falls to non-incremental sync. It happens here https://github.com/bazelbuild/intellij/blob/5ad3c3d5bf3d61b5184c3cc21e230abe994e5c43/base/src/com/google/idea/blaze/base/sync/actions/IncrementalSyncProjectAction.java#L54-L57

------------------

On the other hand when I switch back from QS to AS, it results in Incremental Sync which might cause some unexpected issues (For example leftoveers after QS in project model)

Reply:
```
  private static boolean allTargetsBuild(SyncMode mode) {
    return mode == SyncMode.FULL || mode == SyncMode.INCREMENTAL;
  }
```

With
```
    } else {
      BlazeSyncManager.getInstance(project)
          .incrementalProjectSync(/* reason= */ "IncrementalSyncProjectAction");
    }
```


 
```
  public void incrementalProjectSync(String reason) {
    BlazeSyncParams syncParams =
        BlazeSyncParams.builder()
            .setTitle("Sync")
            .setSyncMode(SyncMode.INCREMENTAL)
            .setSyncOrigin(reason)
            .setAddProjectViewTargets(true)
            .setAddWorkingSet(BlazeUserSettings.getInstance().getExpandSyncToWorkingSet())
            .build();
    requestProjectSync(syncParams);
  }
```
And "Full" is:
```
  public void fullProjectSync(String reason) {
    BlazeSyncParams syncParams =
        BlazeSyncParams.builder()
            .setTitle("Full Sync")
            .setSyncMode(SyncMode.FULL)
            .setSyncOrigin(reason)
            .setAddProjectViewTargets(true)
            .setAddWorkingSet(BlazeUserSettings.getInstance().getExpandSyncToWorkingSet())
            .build();
    requestProjectSync(syncParams);
  }
```

Reply:
TBH I've been using "INCREMENTAL" all this time and there is nothing really incremental about it :)

Reply:
well, this means `INCREMENTAL` was pretty good, theoretically it should always work

Reply:
but I would fall back to FULL, for sure external repo completions in starlark won't work

Reply:
After checking the code I'm not that sure about
> QS is able to detect it's the first sync and falls to non-incremental sync

since nothing resets `com.google.idea.blaze.base.qsync.QuerySyncManager#loadedProject` for QS to do FULL or `onStartup`

Reply:
well, here it ends up with calling `onStartup`

```

it just seems safer to me